src/main/java/service/impl/WriterServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package service.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import service.WriterService;
+
+public class WriterServiceImpl implements WriterService {
+
+    @Override
+    public String writeToFile(String report, String reportPath, String defaultReportPath) {
+        Path filePath;
+        try {
+            filePath = Path.of(reportPath);
+        } catch (Exception e) {
+            System.out.println("Unable to perform operations with destination " + reportPath
+                    + ", switching to default destination");
+            reportPath = defaultReportPath;
+            filePath = Path.of(reportPath);
+        }
+
+        try {
+            Files.deleteIfExists(filePath);
+            Files.write(filePath, report.getBytes());
```

Comment:
```suggestion
            try {
            Path filePath = Path.of(reportPath);
            Files.deleteIfExists(filePath);
            Files.write(filePath, report.getBytes());
```
will this work?

Reply:
Of course will, but my goal was to make extra logic to handle situation, when destination folder is unavailable but there is no other reason not to create report.
So program puts report in default destination.
Leave this please to mentor)

Reply:
I would say that this is also a non-consistent behavior, and we are making a decision instead of the user, he can try to generate a report specifying the path to it, get some exceptions, and not look in the logs of your application. 
So he won't be able to find the report + this functionality was not specified in the requirements, although you did a good job, I think it is better to just throw an exception and have some constant default path in the main logic

Reply:
That should go in loading screen / pop-ups instead of logs, but I don't know how to do it yet)
I'll remove everything extra.